Convert to typescript and simplify API (#11)
kuzzleio/kuzzle-vault: README.md

## About

> Kuzzle Vault offers a secure storage system for secrets. It can encrypt your secrets inside a file and also decrypt in memory the encrypted file.

Reviewer: ```suggestion
Kuzzle Vault offers a secure storage system for secrets. It can encrypt your secrets in a file then easily decrypt & load them into memory.
```

___

## Usage

First, you need to encrypt your secrets. The easiest way to do that is to use Kourou, the Kuzzle CLI.

```bash
$ kourou vault:encrypt config/prod/secrets.json --vault-key <password>

 🚀 Kourou - Encrypts an entire file.
 
 [✔] Secrets were successfully encrypted into the file config/prod/secrets.enc.json
```

Then, you can securely store your secrets inside your repository and share them with you team. 

To load the secrets inside an application, instantiate the Vault with the same password as for the encryption and the path to the secrets file.  

Then, use the decrypt method to load the secrets into the memory.  

```js
const vault = new Vault('password');
vault.decrypt('config/prod/secrets.env.json');

// secrets are now available
vault.secrets
```

You can also provide the password with the environment variable `KUZZLE_VAULT_KEY`.  

```js
// process.env.KUZZLE_VAULT_KEY = 'password'

const vault = new Vault();
vault.decrypt('config/prod/secrets.enc.json');

// secrets are now available
vault.secrets
```

### Strict mode

> In production, you may want to ensure that 

Reviewer 2: incomplete documentation?

Author: I removed this part for now because I though it was too specific.  

This was the part about aborting if there is a key but not password or a password but no key

Reviewer 2: Ok but... this sentence is cut short and is incomplete. As it is, this section doesn't make sense.

Author: (I forgot to push :expressionless: )

## Vault class

[Vault.constructor](#constructor)
[Vault.decrypt](#decrypt)

___

### Vault.constructor

The constructor of the `Vault` class.

```js
Vault(vaultKey: string | undefined);
```

**Arguments**

| Name | Type              | Description |
| -------- | ----------------- | ----------- |
| `vaultKey`  | <pre>String</pre> | The key used to encrypt and decrypt secrets   |

#### Usage

```js
const vault = new Vault('my vault key');
```

___

### Vault.decrypt

> Decrypt the content of the file designated by `encryptedVaultPath` and store the decrypted content inside the property `secrets` of the `Vault` class.

Reviewer 2: ```suggestion
Decrypts the content of the file designated by `encryptedVaultPath` and store the decrypted content inside the property `secrets` of the `Vault` class.
```
